internal/benchrunner/runners/stream/scenario.go

```diff
@@ -0,0 +1,114 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package stream
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/elastic/go-ucfg/yaml"
+)
+
+const devPath = "_dev/benchmark/rally"
+
+type scenario struct {
+	Package     string     `config:"package" json:"package"`
+	Description string     `config:"description" json:"description"`
+	Version     string     `config:"version" json:"version"`
+	DataStream  dataStream `config:"data_stream" json:"data_stream"`
+	Corpora     corpora    `config:"corpora" json:"corpora"`
+}
+
+type dataStream struct {
+	Name string `config:"name" json:"name"`
+}
+
+type corpora struct {
+	Generator *generator `config:"generator" json:"generator"`
+}
+
+type generator struct {
```

Comment:
This seems to be almost the same as https://github.com/elastic/elastic-package/blob/main/internal/benchrunner/runners/rally/scenario.go, same for other objects. Is the duplication on purpose?

Reply:
+1, we should probably try to reduce duplication, there is quite a lot between benchmark runners. This would also help to evaluate better how much logic to maintain we are adding with each runner.

Reply:
yes, let's plan for another PR moving repeated code to some common package under benchmark